src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+        StringBuilder builder = new StringBuilder();
+        String report;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+            String[] lines = builder.toString().split(System.lineSeparator());
+            int supplySum = 0;
+            int buySum = 0;
+            for (String line : lines) {
+                String[] fields = line.split(",");
+                if (fields.length == 2) {
+                    String operation = fields[0].trim();
+                    int amount = Integer.parseInt(fields[1].trim());
+                    if (operation.equals("supply")) {
+                        supplySum += amount;
+                    } else if (operation.equals("buy")) {
+                        buySum += amount;
+                    }
+                }
+            }
+            int result = supplySum - buySum;
 
+            report = "supply," + supplySum + System.lineSeparator()
```

Comment:
Use `StringBuilder` instead of `String` concatenation in loops to improve performance.